stdlib/public/runtime/Metadata.cpp

```diff
@@ -3914,3 +3914,25 @@ bool Metadata::satisfiesClassConstraint() const {
   // or it's a class.
   return isAnyClass();
 }
+
+#if !NDEBUG
+void swift::verifyMangledNameRoundtrip(const Metadata *metadata) {
+  // Disable verification when a special environment variable is set. Some metatypes
+  // crash when going through the mangler or demangler. Tests which trigger those
+  // crashes set this variable to avoid a ton of new test failures. When the tests
+  // are fixed, remove this.
+  static bool verificationDisabled = getenv("SWIFT_DISABLE_MANGLED_NAME_VERIFICATION");
```

Comment:
It doesn't matter too much for a debug-only thing like this, but in general in the runtime we prefer using dispatch_once over C++ static variable initialization since it's a bit lighter-weight. You could use the `SWIFT_LAZY_CONSTANT` macro here:

```
  bool verificationDisabled = SWIFT_LAZY_CONSTANT((bool)getenv("SWIFT_DISABLE_MANGLED_NAME_VERIFICATION"));
```

Reply:
Thanks, I didn't realize this did a better job. Might as well get it right, even if it is a temporary debug-only thing.